### PR TITLE
COMPAT: Update for NumPy dev

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/17986

xref https://github.com/numpy/numpy/pull/9487

Do we want this for 0.21?

### CODE REVIEW

File: pandas/io/packers.py [hunk 1]
```diff
@@ -350,8 +350,11 @@ def unconvert(values, dtype, compress=None):
                 )
                 # fall through to copying `np.fromstring`
 
-    # Copy the string into a numpy array.
-    return np.fromstring(values, dtype=dtype)
+    # Copy the bytes into a numpy array.
+    buf = np.frombuffer(values, dtype=dtype)
+    buf = buf.copy()  # required to not mutate the original data
```
Comment (chris-b1) [inline pandas/io/packers.py]: I _think_ it's safe to not copy here, IIRC `values` is pile of bytes read from a msgpack file, so no risk in mutating?

Comment (TomAugspurger) [reply pandas/io/packers.py]: Without the copy the test failed:

```
        for buf, control_buf in zip(not_garbage, control):
            # make sure none of our mutations above affected the
            # original buffers
>           assert buf == control_buf
E           AssertionError: assert b'\x00\x00\x0...x00\x00@\x8f@' == bytearray(b'\x...00\x008\x8f@')
E             At index 6 diff: 240 != 0
E             Use -v to get the full diff

pandas/tests/io/test_packers.py:690: AssertionError
```

I didn't look closely to see if that's a legitimate problem or not.

Comment (chris-b1) [reply pandas/io/packers.py]: Ah, in that case I would trust the test.

Comment (jreback) [reply pandas/io/packers.py]: you need to do this conditionally for numpy > 1.13 I think

Comment (jreback) [reply pandas/io/packers.py]: cc @llllllllll

Comment (TomAugspurger) [reply pandas/io/packers.py]: The tests on the older NumPy passed, let me make sure zlib is installed and that they weren't skipped.

Comment (TomAugspurger) [reply pandas/io/packers.py]: Pretty sure that https://travis-ci.org/pandas-dev/pandas/jobs/292845756 hit it, so I think we're OK.

**Comment by TomAugspurger** [comment]
Huh, must have deleted that branch accidentally.

**Comment by jreback** [comment]
thanks!

**Pull request merged**